examples/stable_diffusion_xl/gm/data/loader.py

```diff
@@ -78,3 +78,71 @@ def create_loader(
     ds = ds.repeat(epoch_size)
 
     return ds
+
+
+def create_loader_dreambooth(
+    instance_data_path,
+    class_data_path,
+    instance_prompt,
+    class_prompt,
+    rank=0,
+    rank_size=1,
+    train_data_repeat=1,
+    *,
+    dataset_config,
+    per_batch_size,
+    total_step=1000,
+    num_parallel_workers=8,
+    shuffle=True,
+    drop_remainder=True,
+    python_multiprocessing=False,
+):
+    r"""Creates dataloader.
+    Returns:
+        BatchDataset, dataset batched.
+    """
+    dataset = get_obj_from_str(dataset_config["target"])(
+        instance_data_path=instance_data_path,
+        class_data_path=class_data_path,
+        instance_prompt=instance_prompt,
+        class_prompt=class_prompt,
+        train_data_repeat=train_data_repeat,
+        **dataset_config.get("params", dict()),
+    )
+    batch_collate_fn, dataset_column_names = dataset.collate_fn, dataset.dataset_column_names
+    dataset_size = len(dataset)
+    num_step_per_epoch = dataset_size // (per_batch_size * rank_size)
+    epoch_size = math.ceil(total_step / num_step_per_epoch)
+
+    de.config.set_seed(1236517205 + rank)
+    cores = multiprocessing.cpu_count()
+    num_parallel_workers = min(int(cores / rank_size), num_parallel_workers)
+    print(f"Dataloader num parallel workers: [{num_parallel_workers}]")
+    if rank_size > 1:
+        ds = de.GeneratorDataset(
+            dataset,
+            column_names=dataset_column_names,
+            num_parallel_workers=min(8, num_parallel_workers),
+            shuffle=shuffle,
+            python_multiprocessing=python_multiprocessing,
+            num_shards=rank_size,
+            shard_id=rank,
+        )
+    else:
+        ds = de.GeneratorDataset(
+            dataset,
+            column_names=dataset_column_names,
+            num_parallel_workers=min(32, num_parallel_workers),
```

Comment:
`num_parallel_workers=min(8, num_parallel_workers) if rank_size > 1 else min(32, num_parallel_workers),`

for clarity